internal/service/pushbasedlogexport/resource.go

```diff
@@ -0,0 +1,174 @@
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var tfPlan TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigReq := NewPushBasedLogExportReq(&tfPlan)
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfPlan.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.UpdatePushBasedLogConfiguration(ctx, projectID, logExportConfigReq).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	timeout, diags := tfPlan.Timeouts.Update(ctx, defaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigResp, err := WaitStateTransition(ctx, projectID, connV2.PushBasedLogExportApi,
+		retryTimeConfig(timeout, minTimeoutCreateUpdate))
+	if err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	newTFModel, diags := NewTFPushBasedLogExport(ctx, projectID, logExportConfigResp, &tfPlan.Timeouts)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var tfState *TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfState.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.DeletePushBasedLogConfiguration(ctx, projectID).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration delete", err.Error())
+		return
+	}
+
+	deleteTimeout, diags := tfState.Timeouts.Delete(ctx, defaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if err := WaitResourceDelete(ctx, projectID, connV2.PushBasedLogExportApi, retryTimeConfig(deleteTimeout, minTimeoutDelete)); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration delete", err.Error())
+		return
+	}
+}
+
+func (r *pushBasedLogExportRS) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), req.ID)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func retryTimeConfig(configuredTimeout, minTimeout time.Duration) retrystrategy.TimeConfig {
+	return retrystrategy.TimeConfig{
+		Timeout:    configuredTimeout,
+		MinTimeout: minTimeout,
+		Delay:      10 * time.Second,
```

Comment:
as all have constants, i'd do it also for Delay

Reply:
added